shellbase-core/src/main/scala/com/sumologic/shellbase/notifications/PopupNotification.scala

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.sumologic.shellbase.notifications
+
+import java.io.File
+
+class PopupNotification extends ShellNotification {
+  override val name: String = "popup"
+
+  override def notify(title: String, message: String): Unit = {
+    val osascript = new File("/usr/bin/osascript")
```

Comment:
Since this is macOS only you may want to 

1. name the class `MacOSPopupNotification`
1. Check `os.name` before looking for the file.

Reply:
I was thinking that we can support different OS-es by this class in the future, with support added as needed. I probably makes sense to extract OSX-related stuff to a separate method at least.